app/src/org/commcare/connect/network/ConnectNetworkHelper.java

```diff
@@ -81,7 +82,37 @@ private static ConnectNetworkHelper getInstance() {
 
     private static final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
     public static Date parseDate(String dateStr) throws ParseException {
-        return dateFormat.parse(dateStr);
+        Date issueDate=dateFormat.parse(dateStr);
+        return issueDate;
+    }
+
+    public static Date convertUTCToDate(String utcDateString) {
+        SimpleDateFormat utcFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault());
```

Comment:
Nit: Let's make the SimpleDateFormats private static final variables (like line 83). They can easily be reused, and I think they're unfortunately expensive to REPEATEDLY create and destroy.